crates/filter-parser/src/lib.rs

```diff
@@ -946,6 +1001,59 @@ pub mod tests {
         "###
         );
 
+        insta::assert_snapshot!(p(r#"_vectors _vectors EXISTS"#), @r"
+        Was expecting an operation `=`, `!=`, `>=`, `>`, `<=`, `<`, `IN`, `NOT IN`, `TO`, `EXISTS`, `NOT EXISTS`, `IS NULL`, `IS NOT NULL`, `IS EMPTY`, `IS NOT EMPTY`, `CONTAINS`, `NOT CONTAINS`, `STARTS WITH`, `NOT STARTS WITH`, `_geoRadius`, or `_geoBoundingBox` at `_vectors _vectors EXISTS`.
+        1:25 _vectors _vectors EXISTS
+        ");
+        insta::assert_snapshot!(p(r#"_vectors. embedderName EXISTS"#), @r"
+        The vector filter's embedder is invalid.
+        10:30 _vectors. embedderName EXISTS
+        ");
+        insta::assert_snapshot!(p(r#"_vectors .embedderName EXISTS"#), @r"
+        Was expecting an operation `=`, `!=`, `>=`, `>`, `<=`, `<`, `IN`, `NOT IN`, `TO`, `EXISTS`, `NOT EXISTS`, `IS NULL`, `IS NOT NULL`, `IS EMPTY`, `IS NOT EMPTY`, `CONTAINS`, `NOT CONTAINS`, `STARTS WITH`, `NOT STARTS WITH`, `_geoRadius`, or `_geoBoundingBox` at `_vectors .embedderName EXISTS`.
+        1:30 _vectors .embedderName EXISTS
+        ");
+        insta::assert_snapshot!(p(r#"_vectors.embedderName. EXISTS"#), @r"
+        The vector filter has leftover tokens.
+        22:30 _vectors.embedderName. EXISTS
+        ");
+        insta::assert_snapshot!(p(r#"_vectors."embedderName EXISTS"#), @r#"
+        The vector filter's embedder is invalid.
+        30:30 _vectors."embedderName EXISTS
+        "#);
```

Comment:
Two comments on this one. First of all I think for all of these error we should write "The vector filter's emedder **name** is invalid."
Because otherwise it's not immediately obvious which part is wrong here 🤔 

And second comment, I struggled to see what was wrong because we had a closing delimiter. But it was the one of rust, not the one of the embedder name.
People will struggle the same with their filter in a json value so maybe we could instead return a "Missing closing delimitor `"` in embedder name" or something?
What do you think? I'm not sure that's possible with the `delimited` function of nom, though, so you tell me. We won't rewrite the whole parser for that

Reply:
It was possible using error handling and it works like a charm!